### PR TITLE
DD_TAGS env var support

### PR DESCRIPTION
**Pull request by pedro-stanaka**
## Summary

This PR fixes issue #223 and it is something that I am also interested in using. I hope this is the correct implementation for the problem.

### CODE REVIEW

**Comment by pedro-stanaka** [comment]
@hush-hush I know you had self-assigned issue #223 to you, so maybe you want to give me a review here? Thanks in advance!

**Comment by hush-hush** [comment]
Hi @pedro-stanaka ,

Thanks for opening this PR. Adding support for `DD_TAGS` across all our DogStatsD clients is something we're looking into.

The main issue being that when `DD_TAGS` is set for the whole host and the agent/client are running locally it will be picked by both, so we have to be careful. This is especially important when using containers, VMs or k8s deployments. The last thing we want is for the clients or Agent to assign the wrong tags to a metric by default.

We're also looking into aligning the behavior between all clients around this, which is a bit all over the place right now.

In the meantime, clients that want to fetch tags from the env can easily do so using the library constructor. It's not the best but it's an easy workaround.

**Comment by SophieDeBenedetto** [comment]
> Hi @pedro-stanaka ,
> 
> Thanks for opening this PR. Adding support for `DD_TAGS` across all our DogStatsD clients is something we're looking into.
> 
> The main issue being that when `DD_TAGS` is set for the whole host and the agent/client are running locally it will be picked by both, so we have to be careful. This is especially important when using containers, VMs or k8s deployments. The last thing we want is for the clients or Agent to assign the wrong tags to a metric by default.
> 
> We're also looking into aligning the behavior between all clients around this, which is a bit all over the place right now.
> 
> In the meantime, clients that want to fetch tags from the env can easily do so using the library constructor. It's not the best but it's an easy workaround.

I'm not sure I understand why there is a concern about telling the datadog-go SDK to use `DD_TAGS` may conflict with setting `DD_TAGS` at the agent level? In this approach, the datadog-go SDK looks for a `DD_TAGS` environment variable available to the container running the application code using the SDK. Any tags found there are applied to all metrics emitted by the SDK.

There is also an existing option to [configure a containerized datadog-agent running in Docker or similar to have `DD_TAGS` env var](https://docs.datadoghq.com/getting_started/tagging/assigning_tags/?tab=noncontainerizedenvironments) which would tell the agent to apply any such tags to all metrics it sends out to Datadog.

So, the way I understand this is:

* If you set `DD_TAGS` in the application environment, the datadog-go SDK will attach _those_ tags to all metrics it emits
* If you _also_ set `DD_TAGS` at the agent level, the agent will _also_ add _those_ tags to all metrics it emits

In other words, you get both sets of tags added globally.

Is this not desirable?
